lib/game.rb

```diff
@@ -0,0 +1,120 @@
+require 'pry'
+
+class Game
+  attr_accessor :board, :player_1, :player_2
+
+  # Defining WIN_COMBINATIONS constant
+  WIN_COMBINATIONS = [
+    [0,1,2], # top row
+    [3,4,5], # middle row
+    [6,7,8], # bottom row
+    [0,3,6], # left column
+    [1,4,7], # middle column
+    [2,5,8], # right column
+    [0,4,8], # diagonal 1
+    [2,4,6] # diagonal 2
+  ]
+
+  #Initialized a new game with two players and a new board
+  def initialize(player_1=Players::Human.new("X"), player_2=Players::Human.new("O"), board=Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  #Determines the current character by judging the number of plays made
+  def current_player
+    if self.board.turn_count.even?
+      current_player = @player_1
+    else
+      current_player = @player_2
+    end
+  end
+
+  #define a method to check for winning combinations
+  def won?
```

Comment:
Here you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.detect do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]] &&
      @board.taken?(combo[0] + 1)
    end
  end
```